ddtrace/tracer/spancontext.go

```diff
@@ -201,25 +201,17 @@ func (t *trace) push(sp *span) {
 	if t.full {
 		return
 	}
-	tr, haveTracer := internal.GetGlobalTracer().(*tracer)
 	if len(t.spans) >= traceMaxSize {
 		// capacity is reached, we will not be able to complete this trace.
 		t.full = true
 		t.spans = nil // GC
 		log.Error("trace buffer full (%d), dropping trace", traceMaxSize)
-		if haveTracer {
-			tr.config.statsd.Incr("datadog.tracer.traces.dropped", nil, 1)
```

Comment:
I think this is the wrong fix for the race.
Instead, think more about why these are being called while the tracer is "exiting", and the correct moment to do `t.config.statsd.Close()`

Reply:
> I think this is the wrong fix for the race.

Agreed.

I removed this because I think my approach to this is entirely wrong anyway. I just need to atomically increment an integer that I'll include in the tracer. Then the tracer will periodically report and reset the values. This will minimize any performance impact from recording these stats and avoid the race conditions with multiple goroutines accessing the statsd client.